src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,251 @@
-                                                            {ch.name}
-                                                        </a>
-                                                    ) : null}
-                                                </React.Fragment>
-                                            );
-                                        })}
-                                    </ul>
-                                ) : null}
-                            </li>
-                        );
-                    })}
-                </ul>
-            </BrowserRouter>
+    const sideNavList = (
+        <ul
+            {...rest}
+            aria-expanded={hasParent && open}
+            aria-hidden={hasParent && !open}
+            className={sideNavListClasses}>
+            {React.Children.map(children, (child) => {
+                if (React.isValidElement(child)) {
+                    return React.cloneElement(child, {
+                        isSubItem: hasParent,
+                        onItemSelect: onItemSelect,
+                        selected: selectedId === child.props.id,
+                        selectedId: selectedId
+                    });
+                } else {
+                    return child;
+                }
+            })}
+        </ul>
+    );
+
+    if (title && !hasParent) {
+        return (
+            <div
+                className={sideNavHeaderlasses}>
+                <h1 {...titleProps} className='fd-side-nav__title'>
+                    {title}
+                </h1>
+                {sideNavList}
+            </div>
         );
     }
-}
+
+    return sideNavList;
+};
+
 SideNavList.propTypes = {
-    items: PropTypes.array.isRequired,
-    className: PropTypes.string
+    children: PropTypes.element,
+    className: PropTypes.string,
+    hasParent: PropTypes.bool,
+    open: PropTypes.bool,
+    selectedId: PropTypes.string,
+    title: PropTypes.string,
+    titleProps: PropTypes.object,
+    onItemSelect: PropTypes.func
 };
 
 SideNavList.propDescriptions = {
-    items: 'An array of objects with keys \'id\', \'url\', \'name\', \'hasChild\', \'child\', and \'glyph\' setting the attributes of the items.'
+    hasParent: 'Automatically set to **true** when item has a parent `SideNavListItem`',
+    open: 'Set to **true** to mark `SideNavSubList` as open.',
+    selectedId: 'The id of the selected `SideNavListItem`.',
+    onItemSelect: 'Callback function when user selects an item .'
 };
 
-export const SideNavGroup = props => {
-    const { title, children, className, titleProps, ...rest } = props;
+export class SideNavListItem extends React.Component {
+    constructor(props) {
+        super(props);
 
-    const sideNavGroupClasses = classnames(
-        'fd-side-nav__group',
-        className
-    );
+        this.state = {
+            expanded: this.props.expanded ? this.props.expanded : false
```

Comment:
I think we will need a `getDerivedStateFromProps` (previously known as `componentWillReceiveProps`) to keep the internal state in sync since `expanded` is a prop.

Reply:
Corrected in ca228a2